pkg/scale/encode.go

```diff
@@ -230,23 +303,38 @@ func (es *encodeState) encodeArray(in interface{}) (err error) {
 func (es *encodeState) encodeBigInt(i *big.Int) (err error) {
 	switch {
 	case i == nil:
-		err = fmt.Errorf("nil *big.Int")
+		err = fmt.Errorf("%w", errBigIntIsNil)
 	case i.Cmp(new(big.Int).Lsh(big.NewInt(1), 6)) < 0:
 		err = binary.Write(es, binary.LittleEndian, uint8(i.Int64()<<2))
+		if err != nil {
+			err = fmt.Errorf("writing binary for big int %s: %w", i, err)
```

Comment:
I'm not sure what value this brings.  How would this even produce an error unless `es` wasn't initialised correctly with an `io.Writer` implementation?

Reply:
If the user creates a scale encoder with a network (or disk) implementation for io.Writer for example, writing to it might fail. But yeah for memory buffers, it's kinda very unlikely unless you cut your ram with a knife :smile: 

Adding the big int value to the error context can be useful to know what value failed (together with other error wrappings).